router/manage.js

```diff
@@ -1,268 +1,277 @@
+        exec("node " + path_splitRunner + " " + path_class + " " + courseOfferingId, function(err, stdout, stderr) {
+          if (err !== null) {
+            console.log("%s exec splitRunner error: ", filename, err);
+          }
+          /* node utility_scripts/taskInitializer.js <path_to_directory_with_videos> <class_name> */
+          /* adds the videos to the queue to be transcribed */
+          exec("node " + path_taskInitializer + " " + path_class + " " + courseOfferingId, function(err, stdout, stderr) {
+            if (err !== null) {
+              console.log("%s exec taskInitializer error: ", filename, err);
+            }
+            fs.renameSync(path_file, path.join(path_splitted, filename));
+          });
+        });
+      });
+    });
+  });
+
+  response.end();
+});
+
+/* add the instructors to the database */
+router.post('/addInstructors', function (request, response) {
+   var data = request.body.instructors;
+   var instructors = data.split(/[\s,;:\n]+/);
+   var toadd = [];
+   for(instructor in instructors) {
+    if(instructor.match(/\^w+@[a-z]+?\.edu$/)) {
+      toadd.push(instructor);
+    }
+   }
+   console.log(instructors);
+   console.log(toadd);
+   db.addInstructors(toadd, function(err, res) {
+   // client.sadd("instructors", toadd, function(err, res) {
+      if(err) console.log(err);
+   		console.log("added instructors");
+   });
+   response.send(instructors);
+});
+
+
+/* add the students to the database */
+router.post('/addStudents', function (request, response) {
+   var data = request.body.students;
+   var students = data.split(/[\s,;:\n]+/);
+   db.addStudents(students, function(err, res) {
+   // client.sadd("students", students, function(err, res) {
+   		console.log("added students");
+   });
+   response.send(students);
+});
+
+
+/* upload the file of students, then add students to database */
+//var upload = multer({ storage : storage}).single('studentsFile');
+router.post('/UploadStudentsFiles', function (request, response) {
+  var upload = multer({ storage : storage}).any();
+  upload(request, response, function(err) {
+    //console.log(request.files[0].path);
+    var interface = readline.createInterface({
+      input: fs.createReadStream(request.files[0].path)
+    });
+    interface.on('line', function (line) {
+      db.addStudents(line, function(err) {
+        console.log("added student: " + line);
+      })
+    });
+    /* delete the file after adding to database */
+    fs.unlinkSync(request.files[0].path);
+    response.end();
+  });
+});
+
+
+// /* upload the lecture video and segment it into 4-6 minute chunks */
+var storage = multer.diskStorage({
+    destination: function (req, file, cb) {
+        cb(null, '/data/temp')
+    },
+    filename: function (req, file, cb) {
+        cb(null, file.fieldname + '-' + Date.now())
+    }
+})
+
+var upload = multer({ storage: storage })
+var cpUpload = upload.fields([{ name: 'videos', maxCount: 100 }])
+var index = 0;

```

Comment:
This hack won't work.
Imagine you upload two set of files. The counter won't reset. It'll always increase.

Reply:
That's true. I'll have to think of another way then